fvtest/porttest/si.cpp

```diff
@@ -982,8 +982,24 @@ TEST(PortSysinfoTest, sysinfo_test_sysinfo_set_limit_CORE_FILE)
 			}
 			finalSoftLimit = (originalSoftLimit > newHardLimit)? newHardLimit: originalSoftLimit;
 		} else { /* running as root */
-			const uint64_t newHardLimit = 257;
-			rc = omrsysinfo_set_limit(OMRPORT_RESOURCE_FILE_DESCRIPTORS | OMRPORT_LIMIT_HARD, newHardLimit);
+			/* Try setting hard limit below soft limit. This should fail even for root user. */
+			rc = omrsysinfo_set_limit(OMRPORT_RESOURCE_FILE_DESCRIPTORS | OMRPORT_LIMIT_HARD, descriptorLimit);
+			if (0 == rc) {
```

Comment:
This will only fail if `(originalSoftLimit > descriptorLimit)`. Will this condition be always true? If not, we should add a safeguard here.

```suggestion
			if ((0 == rc) && (originalSoftLimit > descriptorLimit)) {
```